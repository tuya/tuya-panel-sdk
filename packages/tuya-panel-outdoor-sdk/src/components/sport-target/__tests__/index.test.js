import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import SportTarget from '../index';

describe('sport-no-target components', () => {
  it('basic render', () => {
    const wrapper = shallow(
      <SportTarget
        curStep={100}
        targetSteps={200}
        targetText={`Today's goal ï½œ ${100}step`}
        percentText="You have achieved your goal"
        themeColor="blue"
        centerView={{
          bgImage: 0,
          children: <Text>123</Text>,
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
