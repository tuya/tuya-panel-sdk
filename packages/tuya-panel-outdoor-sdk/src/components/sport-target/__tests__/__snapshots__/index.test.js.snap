// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sport-no-target components basic render 1`] = `
<View
  style={
    Object {
      "height": 700,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 240,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 24,
            "height": 60,
            "lineHeight": 60,
            "marginBottom": 8,
            "marginTop": 40,
            "textAlign": "center",
            "width": 540,
          },
          Object {
            "color": "blue",
          },
        ]
      }
    >
      Today's goal ï½œ 100step
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 48,
            "textAlign": "center",
            "width": 400,
          }
        }
      >
        You have achieved your goal
        <Text
          style={
            Object {
              "color": "blue",
            }
          }
        >
          50%
        </Text>
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 460,
        "justifyContent": "center",
      }
    }
  >
    <Progress
      andDegree={270}
      backColor="#E5E5E5"
      backStrokeOpacity={1}
      disabled={true}
      endColor={null}
      foreColor="blue"
      foreStrokeOpacity={1}
      gradientId="Progress"
      max={200}
      min={0}
      needMaxCircle={true}
      needMinCircle={false}
      onSlidingComplete={[Function]}
      onValueChange={[Function]}
      pointerEvents="box-only"
      renderCenterView={null}
      scaleHeight={32}
      startColor={null}
      startDegree={135}
      stepValue={10}
      style={
        Object {
          "height": 432,
          "width": 432,
        }
      }
      thumbFill="blue"
      thumbRadius={14}
      thumbStroke="blue"
      thumbStrokeWidth={0}
      value={100}
      x1="0%"
      x2="100%"
      y1="0%"
      y2="0%"
    />
    <CenterView />
  </View>
</View>
`;
